examples/how-to/perform-http-requests/README.md

```diff
@@ -0,0 +1,128 @@
+requests are executed on the client and on all the validators. That means that the client and each
+validator perform the HTTP request independently. The responses must all match (or at least match
+in a quorum of validators) for the block the be confirmed.
+
+If the response varies per request (as a simple example, due to the presence of a "Date" timestamp
+header in the response), the block proposal may end up being rejected by the validators. If there's
+a risk of that happening, the contract should instead call the service as an oracle, and let the
+service perform the HTTP request and return only the deterministic parts of the response.
+
+## HTTP requests using the service as an oracle
+
+The contract may call the service as an oracle. That means that that contracts sends a query to the
+service and waits for its response. The execution of the contract is metered by executed
+instruction, while the service executing as an oracle is metered by a coarse-grained timer. That
+allows the service to execute non-deterministically, and as long as it always returns a
+deterministic response back to the contract, the validators will agree on its execution and reach
+consensus.
+
+In this scenario, the contract requests the service to perform the HTTP request. The HTTP request
+is also executed in each validator.
+
+## Recommendation
+
+It is recommended to minimize the number of HTTP requests performed in total, in order to reduce
+costs. Whenever possible, it's best to perform the request in the client using the service, and
+forward only the HTTP response to the contract. The contract should then verify that the response
+can be trusted.
+
+If there's no way to verify an off-chain HTTP response in the contract, then the request should be
+made in the contract. However, if there's a risk of receiving different HTTP responses among the
+validators, the contract should use the service as oracle to perform the HTTP request and return to
+the contract only the data that is deterministic. Using the service as an oracle is more expensive,
+so it should be avoided if possible.
+
+## Usage
+
+### Setting Up
+
+Before getting started, make sure that the binary tools `linera*` corresponding to
+your version of `linera-sdk` are in your PATH. For scripting purposes, we also assume
+that the BASH function `linera_spawn_and_read_wallet_variables` is defined.
+
+From the root of Linera repository, this can be achieved as follows:
+
+```bash
+export PATH="$PWD/target/debug:$PATH"
+source /dev/stdin <<<"$(linera net helper 2>/dev/null)"
+```
+
+To start the local Linera network:
+
+```bash
+linera_spawn linera net up --with-faucet --faucet-port 8080
+
+# Remember the URL of the faucet.
+FAUCET_URL=http://localhost:8080
+```
+
+We use the test-only CLI option `--testing-prng-seed` to make keys deterministic and simplify our
+explanation.
+
+```bash
+INFO=($(linera wallet request-chain --faucet $FAUCET_URL))
+CHAIN="${INFO1[0]}"
```

Comment:
typo

Reply:
I suspect this file is not tested in CI. You may need to update `readme_test.rs`

Reply:
Thanks! Now that it's running as a test, I realized there's an issue. We need an HTTP server for the test to work :sweat_smile: And we also need to allow that HTTP server to be accessed from the localnet. I'll look into this tomorrow.

Reply:
I suppose the faucet is also an http server :)

Reply:
It needed a specific response for the test to pass, so I created a simple HTTP server for the test.